mobile/examples/object_detections/iOS/ORTObjectDetection/ModelHandler.swift

```diff
@@ -0,0 +1,463 @@
+            ofType: modelFileInfo.extension
+        ) else {
+            print("Failed to get model file path with name: \(modelFilename).")
+            return nil
+        }
+        
+        let sourcePixelFormat = CVPixelBufferGetPixelFormatType(pixelBuffer)
+        assert(sourcePixelFormat == kCVPixelFormatType_32ARGB ||
+                sourcePixelFormat == kCVPixelFormatType_32BGRA ||
+                sourcePixelFormat == kCVPixelFormatType_32RGBA)
+        
+        let imageChannels = 4
+        assert(imageChannels >= inputChannels)
+        
+        let imageWidth = CVPixelBufferGetWidth(pixelBuffer)
+        let imageHeight = CVPixelBufferGetHeight(pixelBuffer)
+        
+        // Preprocess the image
+        let scaledSize = CGSize(width: inputWidth, height: inputHeight)
+        guard let scaledPixelBuffer = preprocess(ofSize: scaledSize, pixelBuffer) else {
+            return nil
+        }
+        
+        // Start the ORT inference environment
+        let env = try ORTEnv(loggingLevel: ORTLoggingLevel.warning)
+        let options = try ORTSessionOptions()
+        try options.setLogSeverityLevel(ORTLoggingLevel.verbose)
+        try options.setIntraOpNumThreads(self.threadCount) // TODO: check if calling the right methods
+        
+        let session = try ORTSession(env: env, modelPath: modelPath, sessionOptions: options)
+        
+        let interval: TimeInterval
+        
+        let inputName = "normalized_input_image_tensor"
+        
+        guard let rgbData = rgbDataFromBuffer(
+            scaledPixelBuffer,
+            byteCount: batchSize * inputWidth * inputHeight * inputChannels
+        ) else {
+            print("Failed to convert the image buffer to RGB data.")
+            return nil
+        }
+        
+        let inputTensor = try ORTValue(tensorData: NSMutableData(data: rgbData),
+                                       elementType: ORTTensorElementDataType.float,
+                                       shape: [1, 300, 300, 3])
+        // Run ORT InferenceSession
+        let startDate = Date()
+        let outputs = try session.run(withInputs: [inputName: inputTensor],
+                                      outputNames: ["TFLite_Detection_PostProcess",
+                                                    "TFLite_Detection_PostProcess:1",
+                                                    "TFLite_Detection_PostProcess:2",
+                                                    "TFLite_Detection_PostProcess:3"],
+                                      runOptions: try ORTRunOptions())
+        interval = Date().timeIntervalSince(startDate) * 1000
+        
+        guard let rawOutputValue = outputs["TFLite_Detection_PostProcess"] else {
+            throw OrtModelError.error("failed to get model output_0")
+        }
+        
+        let rawOutputData = try rawOutputValue.tensorData() as Data
+        
+        guard let outputArr: [Float32] = arrayCopiedFromData(rawOutputData) else {
```

Comment:
is it possible to avoid copying the output data? maybe it doesn't matter so much if we know the data size is small